.github/workflows/check_in_artifact.yml

```diff
@@ -130,20 +135,21 @@ jobs:
           PR_BODY: ${{ inputs.pull_request_body }}
         run: |
           EXISTING_PR="$(gh pr list --state open --base master --head $HEAD_BRANCH_NAME --json 'url' --jq '.[].url' | head -n 1)"
+          EXISTING_CLOSED_PR="$(gh pr list --state closed --base master --head $HEAD_BRANCH_NAME --json 'mergedAt,url' --jq '.[] | select(.mergedAt == null).url' | head -n 1)"
 
           if [ -n "${EXISTING_PR}" ]; then
-            echo "PR already exists ==> ${EXISTING_PR}"
+            echo "Open PR already exists ==> ${EXISTING_PR"
+            echo "Editing with provided title and body..."
+            gh pr edit --title "$PR_TITLE" --body "$PR_BODY"
+            exit 0
+          elif [ -n "${EXISTING_CLOSED_PR}" ] && [ ${{ steps.prep_commit.outputs.branch_exists }}=='true' ]; then
+            echo "Reopening PR... ${EXISTING_CLOSED_PR}"
+            gh pr reopen "${EXISTING_CLOSED_PR}"
+            echo "Editing with provided title and body..."
+            gh pr edit --title "$PR_TITLE" --body "$PR_BODY"
             exit 0
           else
             echo "Creating PR..."
             gh pr create --title "$PR_TITLE" --body "$PR_BODY"
             exit 0
           fi
```

Comment:
The new logic does the following:
* If there is an existing PR, just edit the title and body with what was given as inputs to this action and do nothing else
* Else if there is a closed PR _and the branch for the closed PR was not deleted_, then reopen the PR and edit the title and body. I added the 2nd condition because the reopen command failed on an old run when the branch of the closed PR had been deleted and we had to create a new branch.
* Else, open a new PR with the provided title and body